### DISCUSSION TITLE
BT condition using a ROS2 Service wrapper not available?

### DISCUSSION THREAD

**Discussion by MrTobiasGit** [BehaviorTree/BehaviorTree.CPP#760] [Q&A]
Hi together,
I am using behaviortree_ros2 and I am wondering why bt_action_node.hpp and bt_service_node.hpp inherit from BT::ActionNodeBase however there is no service_node which inherits from BT::ConditionNode. Does this unavailability has a specific reason? Is it planned in the future? 
kind regards, 
Tobi

**Comment by facontidavide** [comment]
YES!

Both these wrappers use an asynchronous interface. 
**Conditions** in BT.CPP must be very fast, synchronous operations and not have any "side effect". 

Definitifely, not the case here

**Reply by MrTobiasGit** [reply]
Ok I think I have understood it:
So as Conditions must be very fast, it’s recommended to check a periodically ROS2 topics (as there is always a value of the last received message that can be checked). 
As actions might take some time to be fullfilled, it’s recommended to use ROS2 Services or even better ROS2 actions (as they behave asynchronous).

**Reply by facontidavide** [reply]
yes
